### ISSUE TITLE
General feedback thread!

### ISSUE THREAD

**Issue by joehoyle** [humanmade/WordPress-Importer#7]
Tried the importer? Let us know if it works!

If you find any major issues, feel free to open new issues; this thread is mainly for non-bug comments and thoughts :smile:

**Comment by jb510** [comment]
I'm trying the importer now on folder full of split up xml files into multisite (in progress, takes a half hour or more normally)

While that runs, to get more people testing quickly, it'd be good to:
1) state clearly it replaces the WordPress Importer and uses the same CLI arguments. (ie. deactivate the old one)
2) If there are different arguments, document them (I haven't looked at the code).
If those are there (readme/wiki) I missed them.

Things I always struggled with using the old importer:
Authors/Users - Reusing existing users (in split XMLs this was either confusing or didn't work as I expected via the UI or CLI.
Media imports were just very flaky.  Running locally right now, but looking forward to testing it on a dev server that can actually pull down media shortly

**Comment by jb510** [comment]
Note: Plugin is network activated, but doesn't show in plugins.php in the sub-sites. Maybe this is deliberate behavior until there is a UI?

**Comment by jb510** [comment]
I'm a little confused by this and haven't tried recreating the bug.

Imported folder of XML files dumped from .com
/xml/sitename-date-.01.xml

Imported into the primary site of a multi site instance locally using:
wp import xml --authors=create

It ran through all the xml files, yay!  It imported all the posts, yay! It seems like it didn't create _any_ users anywhere, boo!

It could be related to me having blanked a bunch of tables (including wp_users) to reset the test site instead of starting from scratch.  I'll try to reproduce tomorrow.

**Comment by rmccue** [comment]
> state clearly it replaces the WordPress Importer and uses the same CLI arguments. (ie. deactivate the old one)

Oh no, I forgot to document this! You actually want `wp wxr-importer import ...`

**Comment by carosioa** [comment]
I am also getting the same error that @jb510 is and his fix let me activate and it appears to work but still getting a php notice:

"Notice: Undefined index: wp_import in /Users/angelo/Documents/htdocs/wp-content/plugins/WordPress-Importer-master/plugin.php on line 38"

Also want to echo the users thing, it is not creating users that don't exist and doesn't seem to be assigning properly even if there is a user with the same username already set up on the site I am importing to. If I set --user=username all the posts import as that user but in general when I am doing imports they have posts from lots of different users so it would be nice if it could match usernames automatically if they are already present.

Great work on this guys, we have been doing a lot of migration work recently and the WP Importer has been a huge pain in my butt, seems like this is a step in the right direction!

**Comment by anteksiler** [comment]
I am trying to import using the below code:

```
$file = get_template_directory() ."/inc/democontent/demo-content.xml";
$importer = new WXR_Importer(array(
    'fetch_attachments' => true,
));
$importer->import($file);
```

But it's not working.

**Comment by rmccue** [comment]
@anteksiler Can you clarify what you mean by "it's not working"? That's not enough to go on unfortunately :)

**Comment by anteksiler** [comment]
@rmccue Thanks for the reply, as far as I can see there are no posts or media being imported.

**Comment by rmccue** [comment]
@anteksiler Is this a public WXR file? Happy to test and debug :)

**Comment by rmccue** [comment]
Oh, actually, you may need a logger instance: try adding `$importer->set_logger( new WP_Importer_Logger_CLI() )` just before your `->import` line

**Comment by anteksiler** [comment]
@rmccue I just tried by adding the above line, but failed to import again. Here is the demo content I am trying to import: http://cl.ly/code/2i3S2O0O2s3a

Also, I am invoking the function using AJAX, not sure if that matters.

**Comment by rmccue** [comment]
Splitting out into #17. :)

**Comment by ssnepenthe** [comment]
Great work!

I just tried it out on a fresh install of WP 4.4-beta4-35735 with the test data from the codex (http://codex.wordpress.org/Theme_Unit_Test) and am getting a couple of warnings. Not 100% sure if the issue is with the importer or the test data so I figured I should post it here.

`[WARNING] Could not find the post parent for "Unicorn Wallpaper" (post #1045)`

The parent post id is **1158**, title **Post Format: Image** and it is not being imported at all. The relevant item: http://pastebin.com/GE4vgKF5

The other warnings are almost certainly from problems with the test data (comments pointing to non-existent users) but I will post them anyway in case I am missing something:

```
[WARNING] Could not find the author for comment #9
[WARNING] Could not find the author for comment #14
[WARNING] Could not find the author for comment #19
[WARNING] Could not find the author for comment #23
``` 

Each of these comments has a comment user id of **24783058**, author email **example@example.org**, and author name **John Doe**, **Joe Bloggs** or **Jane Doe**.

**Comment by logoscreative** [comment]
When importing, all the posts showed up for me, but the categories and tags didn't import along with them. The old importer worked as expected on the same file. Happy to give any details you need—I did the whole process a few times just to make sure it wasn't a fluke.

**Comment by FiddlerStudios** [comment]
So what's the status on this? A GUI would be really helpful. I'm one of many looking for a replacement for the default Importer, which isn't working correctly. But I'm not comfortable with CLI, and so a WP Admin interface is needed. Happy to send bugs, problems, etc, if there's at least a basic GUI for the rest of us to use.

**Comment by altrovideo** [comment]
Hi, thank you for helping us working better :)

After activating in WP 4.2.2 with the old importer plugin activated and with debug="true" in wp.config.php, this appears:  

``
Strict Standards: Redefining already defined constructor for class WXR_Parser_Regex in /var/www/wordpress.dev2/www/wp-content/plugins/wordpress-importer/parsers.php on line 408
Strict Standards: Declaration of WP_Import::bump_request_timeout() should be compatible with WP_Importer::bump_request_timeout($val) in /var/www/wordpress.dev2/www/wp-content/plugins/wordpress-importer/wordpress-importer.php on line 38
Notice: Undefined index: wp_import in /var/www/wordpress.dev2/www/wp-content/plugins/WordPress-Importer-master/plugin.php on line 38
``
without the default importer plugin, after activating your new one , white screen with this msg:

``
Fatal error: Class 'WP_Import' not found in /var/www/wordpress.dev2/www/wp-content/plugins/WordPress-Importer-master/plugin.php on line 33
``

**Comment by ahmadawais** [comment]
@altrovideo 
Actually, the new plugin code depends on the old plugin's class named against WP_Import() at line #33 of `plugin.php` file. 

So, one needs to install the old WordPress Importer plugin before trying to install the new one. 

@rmccue  I think a note about this would go long way :)

**Comment by ahmadawais** [comment]
Just sent a [pull request](https://github.com/humanmade/WordPress-Importer/pull/35) to address the issue of PHP fatal error in the ReadMe.md file.

**Comment by monolux** [comment]
@ahmadawais I have just downloaded this plugin and used it on a fresh default untouched local WordPress installation version 4.5 running on XAMPP and it works out of the box.

I have not had to install the personally dreaded old version of the importer.

Ran the WP install and without doing anything else uploaded WordPress Importer Redux like described and then went on to upload wptest.io test data and it worked flawless out of the box.

Well there is one [tiny issue](https://github.com/humanmade/WordPress-Importer/issues/42) but I am not sure if this is due to WordPress Importer Redux or due to wptest.io test data.

**Comment by ahmadawais** [comment]
@monolux I think @rmccue has merged the two branches.

**Comment by jmcbade** [comment]
Just tried to Import Them Unit Test Data.  I'm using WP 4.5 and I am on a localhost using MAMP.  I click to start and I get a status screen but never any progress.

**Comment by ahmadawais** [comment]
^Same here.

**Comment by rumspeed** [comment]
I was finally able to test the importer. Overall it is much better than the standard importer and I love the direction it is going and hope I can help out further. Anyway, here are my notes:

Environment: Both the site I created the export from and site I used to run the import were WP v4.5.2. The import site was a fresh install of WP without any content or plugins that create custom post types. So, no new registered post types.

**Prior to import**
- Love the drag and drop area for accepting the export file.
- I'm digging the Import Summary
- I found the facts about sea lions interesting but not relevant. :)

**During Import**
- It sat on "Now Importing" for at least 60 seconds. Then I opened the dashboard in another tab and it started. Not sure if that was related or just a coincidence.
- Seemed to proceed normally after that. Maybe I just needed to be a bit more patient.

**Import Status**
- Import Complete at 89% (see image)
![import_complete_at_89](https://cloud.githubusercontent.com/assets/4145412/15130733/8d480d02-15ff-11e6-81ce-b853545df3fb.png)

**Verifying Import**

1. Pages/Posts transferred 100%
- Status indicated 84/102 transferred. It skipped CPTs that are not registered which is nice because I didn't actually want them. I didn't know it skipped them until I scrolled down through the message log. The UI was a bit confusing because it looked like it didn't actually complete. Maybe a yellow indicator in the status bar for the portion of skipped items. That would fill in the status bar and kick up the percent complete to 100%.
- After reviewing the log, it explains the warning: "warning Failed to import "Welcome": Invalid post type wp-help". Again, didn't see that until I scrolled way down to see it.
- In addition to posts and pages, the site had 4 Slides, 8 Reviews, and 6 Help Documents. That accounts for all 18 of the missing posts. Having a breakdown of the posts for each CPT would be helpful. Also, some indication of why they were skipped (18 records skipped, see warnings in the log below for details. Again, I like that it skipped them but the final status indicator was at 89% when "Import complete!" was displayed.
- Media in posts/pages: Images display properly on posts and the image path is changed following the new site URL. (This was a major pain point from the WP Importer)
- Media: one file missing. It was a 3MB SWF. Our site is set at 12MB max. Likely the file type wasn't whitelisted for this site?
- warning	Could not find the post parent for "Screen Shot 2013-10-24 at 2.16.27 PM" (post #282)
- warning	Could not update "Patient &#038; Family Services" (post #56) with mapped data

**Comment by pipdig** [comment]
I've been waiting to try this out once the GUI was created. I just tried it now and I love it! Even my new found sea lion knowledge :)

The only issue I had was that the importer hung at 99% on 2 remaining term imports http://imgur.com/5NljSNz. From what I can tell, the rest of the import was completely successful, but there were three comment warnings:

warning	Could not find the author for comment #2
warning	Could not find the author for comment #3
warning	Could not find the author for comment #5

I was testing with an export created from a "real" wp.com blog with 125mb media. WP4.5.2 on XAMP

**Comment by Flowgram** [comment]
Looks promising :) Great to see this under development.

I imported: 39 posts, 138 media items (over 400MB), 207 users.

I was importing to the main site of a multisite install.

My issues:
- Post categories didn't come in at all.
- None of the posts got attached to users. Got many "Could not find the author for" errors.
- Some of embedded images contained the new correct src URL. Many still had the old one though.
- When you import again, it should match users in the select drop-down, if they already exist.

Looking forward to seeing this develop.

**Comment by asadkn** [comment]
Just wondering if the web UI was tested on IE/Edge. I noticed it uses EventSource/SSE which isn't available even in the latest Edge version. I didn't find a polyfill under assets. Did I miss something?

**Comment by mantismamita** [comment]
Much more informative than the original :) I think there is a huge need for this too as most WordPress Importers have a drag and drop interface (which is impractical when testing needs to be done in several environments) and are often limited in features. Its also fantastic to be able to use via the command line. 

I did encounter several problems though mainly with not being able to delete posts and re-import them during testing. I tested on a multisite using multi-network. I agree that matching authors would be a big plus and also a bit more debug information (especially concerning the image import)

**Comment by realgeek** [comment]
I got "Could not find the author for" on all the posts and no categories were assigned. I'm using a site with existing users and categories but no content.

WP-CLI 0.24.1
PHP 5.5.9-1ubuntu4.19 (cli) (built: Jul 28 2016 19:31:33)
sudo -u www-data wp wxr-importer import /tmp/output002.wxr
importer plugin downloaded today (2016-09-14)

**Comment by vickinssl** [comment]
Okay, I used this plugin to import a blog with 2500+ posts and 2300+ media attachments into multisite. 

First try, I did a trial run and imported it on my development server, and I let it import the users along with the posts.

Second experience, on the operational site, I already had existing users (we are changing over from typical WP-generated user accounts to user accounts generated through our local LDAP with a plugin), so rather than importing the users, I assigned the posts to the correct new user accounts on the system. Everything else was the same. 

IT WORKED PERFECTLY BOTH TIMES. OH MY GOD THANKYOUTHANKYOUTHANKYOU. Where do I send your pony, man? You guys definitely earned it.

The only oddity I saw was for some reason the importer seemed to think I had 79 terms when in fact there were only 20 categories and 19 tags defined on the blog. On the first import it told me it only imported 40 of the 79 (close enough); the second import it told me it imported zero of them! But in fact, all of the categories and tags were imported and assigned correctly.

Thanks again, you totally rescued my week.

**Comment by cameron707** [comment]
It seemed to hang at the start at 0% but suddenly a whole lot of media items appeared to be uploaded. Not a big deal for those who are patient. Noticing it hang at 15% now as well, with little to no network activity while it's hanging. Hangs for a few minutes on no network activity. There are some files that were already uploaded via previous attempts using the v1 importer. Unsure if that could mess with it. Went from 15% to 87% rather quickly in a matter of seconds. The only warnings I got were some "Could not find post parent" warnings. I think it finished because it says "Import complete" but still the progress is at 87%.

**Comment by Siggytron** [comment]
Hi. I'm VERY new to Wordpress but my first assignment at my new job is to migrate some wordpress sites to a localhost. The database was brand new - 4.7.2 - and empty until I started trying to import. I had problems migrating the media files but I don't know enough about wordpress to know why. I had been trying some work-arounds described in forums when I stumbled upon the Importer Redux. I'm running it now. I'll make another post when it's finished but it is taking a LONG time, so I thought I'd post a couple screenshot images. So far, it's been running an hour. You'll see in one image where I added a yellow arrow, it says the import is complete but there is still quite a ways to go. (I have my fingers crossed that the media will import.) So far it seems that the users have not imported. I've been reading other people's comments and it seems perhaps that I should have created the users in my localhost database before starting the import but I'm so new to wordpress I didn't realize that was a necessary step. (Is it supposed to be a necessary step?) The other image is of a warning I received at the top of my screen. I don't know what it means but here it is.
![wpimporterredux-importsummary](https://cloud.githubusercontent.com/assets/25599683/22675358/3ccd82a4-ecb3-11e6-9fd1-59579a8393a6.jpg)
![wpimporterredux-warning](https://cloud.githubusercontent.com/assets/25599683/22675374/4dff7686-ecb3-11e6-9005-ed479f6d0b35.jpg)

**Comment by Siggytron** [comment]
I'll just say as a follow up that I clicked "Visit Website" because I wanted to see what it looked like and it seems I may have stopped the importer from running. On the other hand, I think the importer may have stopped running even though some of the bars had not reached 100%. I'm not sure. A lot of the media files uploaded but the top menu and title image did not - I guess that was what the warning meant. 

Still, this is a big improvement from the regular importer which did not import ANY media files. There were also some widgets (Is that what they're called?) in that didn't load. There was an Indiegogo box and a Twitter box and a subscribe to our newsletter box. Again, I'm so new to wordpress I'm not even sure if those are widgets or custom post types. Anyway, they didn't appear. 

I hope some of this information is helpful. Good luck on the effort! Gambatte Ne!

**Comment by snyggapa** [comment]
Stock importer just fails with timeout so anything here is an improvement. 

This managed to import all of the media (although counts seem a bit odd - header claims 899 media items	898/899 whereas the log shows 1403 media items.

What it didn't do however is import any pages - is this intentional? The site I am moving all all page based, no posts at all. will open an issue

Thanks

**Comment by BeefKnuckleback** [comment]
I just tried this out on a fairly straightforward blog migration, containing data that the standard importer had to import in tiny little chunks. All posts, pages, users, and media imported in a single import.  Everything looks exactly like it should up until the media library hits an image that's too big for the server to get a handle on (a 5 meg JPEG); from that point forward all imported media is listed as "unattached."

That glitch aside, the plugin seems to work exactly as intended. 👍

**Comment by pandabrand** [comment]
I need some help! I am trying to import content from an expression engine site, so I am making a .wxr file and using that to import. Using the standard importer plugin, I could get my posts in but no media attachments. I am now trying to use this on and things are worse. None of my categories, tags, or authors are importing *AND* the media attachments aren't importing. Is there a place where what the correctly formatted .wxr exists? I have about three different versions, I don't understand completely how attachments are supposed to be formatted what as far what is the purpose of the link, guid, and the attachment_url. I'll start digging in the code here but if there is any documentation outside of this(I can't find any) it would very helpful. 

I guess I'm not even sure if this plugin is made to handle manually made .wxr files either?

**Comment by shivapoudel** [comment]
@rmccue Looks promising but is there any promise for the release too :)

**Comment by ClaudiuCeia** [comment]
Any idea on the release date for a "public beta"? It would be nice to have this available via `wpackagist`, which requires it to be listed on wordpress.org.

**Comment by drzraf** [comment]
non-bug comment: please don't abandon the project before it get merged into core. Merge (or at least give a look/comment) at the stack of PR

**Comment by iamkingsleyf** [comment]
Looks promising but has not been updated for the last 2YEARS.

**Comment by TheRefillMan** [comment]
Thank you, so much for your importer, Joe.  It addressed the 504 Gateway timeout error I had been wrestling with for the last 24 hours (something six other more "whiz-baang" "up-to-date" importers didn't do!)  Thank you, again!

**Comment by ricjcs** [comment]
This plugin saved me. All the content was imported even though it has some problems to solve.

I needed to move a wordpress.com blog with more than 600 articles and more than 1000 images. The current WordPress Import was failing to import. I divided the xml files but still had trouble importing the content.

I searched for solutions on the internet and found this plugin that successfully imported the content.

**The problems:**
- Not all menus were working, I had to redo them.
- In the categories section and the tag section, some of them did not show the associated posts. Solution: updating all posts in bulk resolves.
- I had about 600 detached images.

These were the problems that remained to be solved, so far I haven't detected anything else, and **all the content has been imported**.

Too bad that this plugin has stopped developing, it is very promising! Thanks for the importer!

**Comment by ricjcs** [comment]
I found 2 more problems:

- Some images in the gallery have the url: files.wordpress.com/
- The images in the cover blocks also had the url: files.wordpress.com/

**Comment by dannyfoo** [comment]
I couldn't find an FAQ on what this meant. And though it was mentioned above, I couldn't find an explanation of why this warning occurred. So, I hope someone could shed light on this.

> warning | Could not find the post parent for "Doctor" (post #35814)

Does this mean:
A ) The post does not exist ?
B ) The media named; Doctor, could not be attached to the post #35814 ?
C ) Something else ?

Thanks in advance.

**Comment by francescocarlucci** [comment]
Hello Folks,

I hope this is the proper place to discuss this as the repo doesn't have a security policy :)

I believe that wp_safe_remote_get() should be used in place of wp_remote_get() on https://github.com/humanmade/WordPress-Importer/blob/140a53eb597de87d786cf942c8fe74fc97432f14/class-wxr-importer.php#L1774 inside fetch_remote_file()

That function is designed to make HTTP requests to sources that may be untrusted, so it's by design prone to SSRF. Plus, many plugins are incorporating the library as a dependency, so it would be wise to retrieve remote content using a function that has built-in protection against SSRF.

What do you think?

Cheers,
Francesco
